src/system/PageLockModule/Api/LockingApi.php

```diff
@@ -0,0 +1,262 @@
+ * file that was distributed with this source code.
+ */
+
+namespace Zikula\PageLockModule\Api;
+
+use PageUtil;
+use Twig_Environment;
+use Zikula\PageLockModule\Entity\PageLockEntity;
+
+/**
+ * Class LockingApi.
+ * @package Zikula\PageLockModule\Api
+ *
+ * This class provides means for using a locking mechanism.
+ * It should be used instead of the old user api.
+ */
+class LockingApi
+{
+    /**
+     * length of time to lock a page
+     */
+    const PAGELOCKLIFETIME = 30;
+
+    /**
+     * @var Twig_Environment
+     */
+    private $twig;
+
+    /**
+     * LockingApi constructor.
+     *
+     * @param Twig_Environment $twig Twig service instance.
+     */
+    public function __construct(Twig_Environment $twig)
+    {
+        $this->twig = $twig;
+    }
+
+    /**
+     * Add the page locking code to the page header
+     *
+     * @param string $lockName        The name of the lock to be released
+     * @param string $returnUrl       The URL to return control to (optional) (default: null)
+     * @param bool   $ignoreEmptyLock Ignore an empty lock name (optional) (default: false)
+     *
+     * @return bool true
+     */
+    public function pageLock($lockName, $returnUrl = null, $ignoreEmptyLock = false)
+    {
+        $hasLock = true;
+        $lockedHtml = '';
+
+        if (!empty($lockName) || !$ignoreEmptyLock) {
+            PageUtil::addVar('javascript', 'zikula.ui');
+            PageUtil::addVar('javascript', 'system/PageLockModule/Resources/public/js/PageLock.js');
+            PageUtil::addVar('stylesheet', \ThemeUtil::getModuleStylesheet('ZikulaPageLockModule'));
+
+            $lockInfo = $this->requireLock($lockName, \UserUtil::getVar('uname'), $_SERVER['REMOTE_ADDR']);
+
+            $hasLock = $lockInfo['hasLock'];
+            if (!$hasLock) {
+                $templateParameters = [
+                    'lockedBy' => $lockInfo['lockedBy']
+                ];
+                $lockedHtml = $this->twig->render('@ZikulaPageLockModule/lockedWindow.html.twig', $templateParameters);
+            }
+        }
+
+        $html = "<script type=\"text/javascript\">/* <![CDATA[ */ \n";
+        $html .= "( function($) {\n";
+
+        if (!empty($lockName)) {
+            if ($hasLock) {
+                $html .= "    $(document).ready(PageLock.UnlockedPage);\n";
+            } else {
+                $html .= "    $(document).ready(PageLock.LockedPage);\n";
+            }
+        }
+
+        $html .= "})(jQuery);\n";
+
+        // Use "self::PAGELOCKLIFETIME*2/3" to add a good margin to lock timeout when pinging
+
+        // disabled due to #2556 and #2745
+        // $returnUrl = \DataUtil::formatForDisplayHTML($returnUrl);
+
+        $html .= "
+PageLock.LockName = '$lockName';
+PageLock.ReturnUrl = '$returnUrl';
+PageLock.PingTime = " . (self::PAGELOCKLIFETIME * 2 / 3) . ";
+ /* ]]> */</script>";
+
+        PageUtil::addVar('header', $html);
```

Comment:
scripts should be added to footer, right? and wrapped in standard:

        (function($) {
            $(document).ready(function() {
                 // script stuff...
            });
        })(jQuery);